MoreLinq/HasDuplicates.cs

```diff
@@ -0,0 +1,59 @@
+
+        /// <summary>
+        ///   Checks if sequence contains duplicates, using the specified element equality comparer
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <param name="comparer">The equality comparer to use to determine whether or not keys are equal.
+        /// If null, the default equality comparer for <c>TSource</c> is used.</param>
+        /// <typeparam name="T">The type of the elements in the source sequence</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<T>(this IEnumerable<T> source, IEqualityComparer<T>? comparer)
+            => source?.HasDuplicates(IdFn, comparer) ?? throw new ArgumentNullException(nameof(source));
+
+        /// <summary>
+        ///   Checks if sequence contains duplicates, using the specified element equality comparer
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <param name="keySelector">Projection for determining "distinctness"</param>
+        /// <typeparam name="TSource">Type of the source sequence</typeparam>
+        /// <typeparam name="TKey">Type of the projected element</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector)
+            => source?.HasDuplicates(keySelector, EqualityComparer<TKey>.Default) ?? throw new ArgumentNullException(nameof(source));
+
+        /// <summary>
+        ///   Checks if sequence contains duplicates, using the specified element equality comparer
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <param name="keySelector">Projection for determining "distinctness"</param>
+        /// <param name="comparer">The equality comparer to use to determine whether or not keys are equal.
+        /// If null, the default equality comparer for <c>TSource</c> is used.</param>
+        /// <typeparam name="TSource">Type of the source sequence</typeparam>
+        /// <typeparam name="TKey">Type of the projected element</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector, IEqualityComparer<TKey>? comparer)
+        {
+            if (source is null) throw new ArgumentNullException(nameof(source));
+            if (keySelector is null) throw new ArgumentNullException(nameof(keySelector));
+
+            var enumeratedElements = new HashSet<TKey>(comparer);
+            return source.Any(element => !enumeratedElements.Add(keySelector(element)));
```

Comment:
maybe should be better to reimplement it with imperative way using a foreach.. not using `Any` and lambdas would use less resources, what is a great thing for all users of a shared library.